### ISSUE TITLE
Ideas to Improve the searchability & browsability aspects of The Turing Way Book (ISCAS 2021)

### ISSUE THREAD

**Issue by Spnetic-5** [alan-turing-institute/the-turing-way#1966]
This issue addresses #1787 

### Summary
*Hello I'm Saurabh, I'm planning to contribute to the 'The Turing Way' project in ISCAS 2021.* This issue is part of the task for the interested candidates for the 'The Turing Way' project.

Here are some of the ideas I have referred, thought of :
* ### What needs to be done?

- [ ] *Context menu column on the left side can be made more readable like [this](https://jupyterbook.org/interactive/hiding.html).*
- [ ] *advanced responsive search box (that suggests searches) and adding an option to filter the contents.*
- [ ] *Adding alphabetical order to the chapters like [Deep learning for Molecules and Materials](https://whitead.github.io/dmol-book/intro.html#)*.
- [ ]  *connect the book with online services like Binder, Google Colab.*
- [ ]  *make the welcome page more intuitive for new users by highlighting keywords(bold), adding more bullet points.*
- [ ]  *using github badges in the book as a link to different repo sections like [this](https://openpifpaf.github.io/intro.html)*.
- [ ]  *clickable options to navigate different personas*.
- [ ]  *adding more colors to the content for a better user interface like [FAIR cookbook](https://fairplus.github.io/the-fair-cookbook/content/recipes/introduction/FAIR-cookbook-audience.html).*
- [ ]  *adding different meta tags in HTML head to improve SEO.*
- [ ]  *summarizing, concluding each page's content in a single line at the end.*
- [ ]  *adding more styling to the previous/next buttons, long text paras, headings, and subheadings using CSS.*
- [ ]  *making TOC sidebar also collapsable like the context menu bar on left.*

Please do let me know if I did not miss anything or any suggestions.

**Comment by BatoolMM** [comment]
Thank you @Spnetic-5 for listing all these ideas - some of these suggestions have also been discussed previously in #1960.

We are very keen to add clickable options to navigate different personas and also add different meta tags in HTML head to improve SEO. It would be nice if you can add more details of how you might approach either of these ideas or submit a very small experimental PR?  

**Thank you again for submitting all these interesting ideas.**

**Comment by Spnetic-5** [comment]
Yes, @BatoolMM, I'm aware of #1960, I would definitely work on a detailed proposal on how I would be approaching the implementation. 
*  I would like to have your suggestion on what would be better:
 * [ ]  1. Should I first make a detailed proposal first and then start implementing it step by step. Or
 * [ ]  2. Should I first start working on small experimental PR?

**Comment by BatoolMM** [comment]
Awesome @Spnetic-5 ! 
I'd recommend submitting a very small experimental PR but you are also welcome to start with a detailed proposal of how you might want to implement these ideas if it's more convenient for you. 

Please feel free to ask me if you have any questions.

**Comment by Spnetic-5** [comment]
I will send the small experimental PR soon and also start with my proposal, Thanks for the response!

**Comment by Spnetic-5** [comment]
Hello @BatoolMM, @martinagvilas, I have done little work on the Caption bar of the TW book in order to make it more readable and planning to send PR for the same, could you please review it and give me some suggestions!

This is a demo implementation of 3rd task mentioned in https://github.com/alan-turing-institute/the-turing-way/issues/1966#issue-912731664

- [X] *added alphabetical order to caption bar.*
- [X] *made heading of each topic **bold**.*

* Before: 
![alt-text-1](https://user-images.githubusercontent.com/66636289/120941487-a41b0280-c740-11eb-80ca-c9c2483e2f44.png)

* After:
![alt-text-2](https://user-images.githubusercontent.com/66636289/120941549-ddec0900-c740-11eb-90cb-640506d11784.png)

**Comment by Spnetic-5** [comment]
@BatoolMM Please give some reviews on this!

**Comment by BatoolMM** [comment]
As I mentioned previously in #1960, adding alphabetical order and numbering the sections might not help much, especially if the TOC is collapsable. Can your experimental PR focus on either persona-based exploration or adding different meta tags in HTML head to improve SEO?

**Comment by Spnetic-5** [comment]
Yes @BatoolMM, I'd prefer to work on an experimental PR to boost SEO by adding various meta tags to the HTML head. At the same time, I'll consider an approach for persona-based navigation.
